integration_test/third_party_apps_test.go

```diff
@@ -187,6 +190,34 @@ type expectedEntry struct {
 	FieldMatchers map[string]string `yaml:"field_matchers"`
 }
 
+// expectedMetricsEntry encodes a series of assertions about what data we expect
+// to see in the metrics backend.
+type expectedMetricEntry struct {
+	// The metric type, for example workload.googleapis.com/apache.current_connections.
+	Type string `yaml:"type"`
+	// The value type, for example INT64.
+	ValueType string `yaml:"value_type"`
```

Comment:
There's only a limited set of those. Should we add validation here?

Ditto for `Kind` and `MonitoredResource`…

Reply:
Added YAML validation for all three.